lib/games_puzzles_algorithms/games/ttt/game_state.py

```diff
@@ -72,6 +72,137 @@ def __iter__(self):
         return (i for i in self._data)
 
 
+class WinDetector(object):
+    """
+    The WinDetector is used to track the win state of a game of tic-tac-toe.
+    This is done by tracking run-lengths on the board for each direction. We
+    perform this count from left to right and top to bottom along the board.
+    The value in any given cell is the length of the run ending in that cell.
+    """
+    def __init__(self, num_rows, num_columns, spaces_to_win):
```

Comment:
Thanks for prefixing `rows` and `columns` with `num`, but maybe do the same for `spaces_to_win`?